lucene/core/src/java/org/apache/lucene/codecs/lucene80/Lucene80DocValuesProducer.java

```diff
@@ -0,0 +1,1413 @@
+          set = false;
+          return disi.advance(target);
+        }
+
+        @Override
+        public boolean advanceExact(int target) throws IOException {
+          set = false;
+          return disi.advanceExact(target);
+        }
+
+        @Override
+        public long nextOrd() throws IOException {
+          if (set == false) {
+            final int index = disi.index();
+            final long start = addresses.get(index);
+            this.start = start + 1;
+            end = addresses.get(index + 1L);
+            set = true;
+            return ords.get(start);
+          } else if (start == end) {
+            return NO_MORE_ORDS;
+          } else {
+            return ords.get(start++);
+          }
+        }
+
+      };
+    }
+  }
+
+  @Override
+  public void checkIntegrity() throws IOException {
+    CodecUtil.checksumEntireFile(data);
+  }
+
+  /**
+   * Reader for longs split into blocks of different bits per values.
+   * The longs are requested by index and must be accessed in monotonically increasing order.
+   */
+  // Note: The order requirement could be removed as the jump-tables allow for backwards iteration.
+  private class VaryingBPVReader {
+    final RandomAccessInput slice;
+    final NumericEntry entry;
+    final int shift;
+    final long mul;
+    final int mask;
+
+    long block = -1;
+    long delta;
+    long offset;
+    long blockEndOffset;
+    LongValues values;
+
+    VaryingBPVReader(NumericEntry entry) throws IOException {
+      this(entry, data.randomAccessSlice(entry.valuesOffset, entry.valuesLength));
+    }
+    VaryingBPVReader(NumericEntry entry, RandomAccessInput slice) throws IOException {
+      this.entry = entry;
+      this.slice = slice;
+      shift = entry.blockShift;
+      mul = entry.gcd;
+      mask = (1 << shift) - 1;
+//      System.out.println("vBPB: Reader created with underlying data(offset=" + data.getFilePointer() + ", length=" +
+//          data.length() + "), slice(offset=" + entry.valuesOffset + ", length=" + entry.valuesLength +
+//          ", total=" + (entry.valuesOffset + entry.valuesLength) + ")");
+    }
+
+    long getLongValue(long index) throws IOException {
+      final long block = index >>> shift;
+      if (this.block != block) {
+        int bitsPerValue;
+        do {
+          // If the needed block is the one directly following the current block, it is cheaper to avoid the cache
+          if (entry.jumpTableOffset != -1 && block != this.block+1) {
+//            System.out.println("Jumptabling to index=" + index +
+//                ", current(offset=" + data.getFilePointer() + ", length=" + data.length() +
+//                "), jumpTable(offset=" + entry.jumpTableOffset + ", jumpTableOffset=" + block*Long.BYTES + ", absolute=" +
+//                (entry.jumpTableOffset+block*Long.BYTES) + ")");
+            blockEndOffset = slice.readLong(entry.jumpTableOffset+block*Long.BYTES-entry.valuesOffset)-entry.valuesOffset;
```

Comment:
This one is reading alternatively from the jump table and from actual data too. Should we have two slices?